blots/scroll.ts

```diff
@@ -242,6 +324,123 @@ class Scroll extends ScrollBlot {
       blot.updateContent(change);
     }
   }
+
+  private deltaToRenderBlocks(delta: Delta) {
+    const renderBlocks: RenderBlock[] = [];
+
+    let currentBlockDelta = new Delta();
+    delta.forEach(op => {
+      const insert = op?.insert;
+      if (!insert) return;
+      if (typeof insert === 'string') {
+        const splitted = insert.split('\n');
+        splitted.slice(0, -1).forEach(text => {
+          currentBlockDelta.insert(text, op.attributes);
+          renderBlocks.push({
+            type: 'block',
+            delta: currentBlockDelta,
+            attributes: op.attributes ?? {},
+          });
+          currentBlockDelta = new Delta();
+        });
+        const last = splitted[splitted.length - 1];
+        if (last) {
+          currentBlockDelta.insert(last, op.attributes);
+        }
+      } else {
+        const key = Object.keys(insert)[0];
+        if (!key) return;
+        if (this.query(key, Scope.INLINE)) {
+          currentBlockDelta.push(op);
+        } else {
+          if (currentBlockDelta.length()) {
+            renderBlocks.push({
+              type: 'block',
+              delta: currentBlockDelta,
+              attributes: {},
+            });
+          }
+          currentBlockDelta = new Delta();
+          renderBlocks.push({
+            type: 'blockEmbed',
+            key,
+            value: insert[key],
+            attributes: op.attributes ?? {},
+          });
+        }
+      }
+    });
+
+    if (currentBlockDelta.length()) {
+      renderBlocks.push({
+        type: 'block',
+        delta: currentBlockDelta,
+        attributes: {},
+      });
+    }
+
+    return renderBlocks;
+  }
+
+  private createBlock(attributes: AttributeMap) {
+    let blotName: string | undefined;
+    const formats: AttributeMap = {};
+
+    Object.entries(attributes).forEach(([key, value]) => {
+      const isBlockBlot = this.query(key, Scope.BLOCK & Scope.BLOT) != null;
+      if (isBlockBlot) {
+        blotName = key;
+      } else {
+        formats[key] = value;
+      }
+    });
+
+    const block = this.create(
+      blotName || this.statics.defaultChild.blotName,
+      blotName ? attributes[blotName] : undefined,
+    ) as ParentBlot;
+
+    const length = block.length();
+    Object.entries(formats).forEach(([key, value]) => {
+      block.formatAt(0, length, key, value);
+    });
+
+    return block;
```

Comment:
Changing to:

```js
    return block.statics.requiredContainer
      ? block.wrap(block.statics.requiredContainer.blotName)
      : block;
```

Decreases the rendering time for /stress by 500ms but not super sure if it's safe so this PR leaves it out.